asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py

```diff
@@ -0,0 +1,170 @@
+    factory_file: The name of the JSON file with the configured AlchemyPrepWorkflow, if not supplied the default will be
+        used but a core smarts must be provided.
+    core_smarts: The SMARTS string used to identify the atoms in each ligand to be constrained. Required if the factory file is not supplied.
+    """
+    from asapdiscovery.data.openeye import save_openeye_sdfs
+    from asapdiscovery.data.schema_v2.complex import PreppedComplex
+    from asapdiscovery.data.schema_v2.molfile import MolFileFactory
+
+    pretty.install()
+    console = rich.get_console()
+    print_header(console)
+
+    # load the factory and set the core smarts if supplied
+    if factory_file is not None:
+        factory = AlchemyPrepWorkflow.parse_file(factory_file)
+        if core_smarts is not None:
+            factory.core_smarts = core_smarts
+    else:
+        factory = AlchemyPrepWorkflow(core_smarts=core_smarts)
+
+    # load the molecules
+    asap_ligands = MolFileFactory.from_file(filename=ligands).ligands
+
+    message = Padding(
+        f"Loaded {len(asap_ligands)} ligands from [repr.filename]{ligands}[/repr.filename]",
+        (1, 0, 1, 0),
+    )
+    console.print(message)
+
+    # TODO make a general from_file which handles this
+    if ".json" in receptor_complex:
+        ref_complex = PreppedComplex.parse_file(receptor_complex)
+    else:
+        receptor_complex = PreppedComplex.from_oedu_file(receptor_complex)
+
+    message = Padding(
+        f"Loaded a prepared complex from [repr.filename]{receptor_complex}[/repr.filename]",
+        (1, 0, 1, 0),
+    )
+    console.print(message)
+
+    message = Padding("Starting Alchemy-Prep workflow", (1, 0, 1, 0))
+    console.print(message)
+
+    alchemy_dataset = factory.create_alchemy_dataset(
+        dataset_name=dataset_name, ligands=asap_ligands, reference_complex=ref_complex
+    )
+    output_folder = pathlib.Path(dataset_name)
+    output_folder.mkdir(parents=True, exist_ok=True)
+
+    dataset_file = output_folder.joinpath("prepared_alchemy_dataset.json")
+    alchemy_dataset.to_file(dataset_file)
+    message = Padding(
+        f"Saved AlchemyDataset to [repr.filename]{dataset_file}[/repr.filename]",
+        (1, 0, 1, 0),
+    )
+    console.print(message)
+
+    posed_ligand_file = output_folder.joinpath("posed_ligands.sdf")
+    alchemy_dataset.save_posed_ligands(posed_ligand_file)
+    message = Padding(
+        f"Saved posed ligands to [repr.filename]{posed_ligand_file}[/repr.filename]",
+        (1, 0, 1, 0),
+    )
+    console.print(message)
+
+    if alchemy_dataset.failed_ligands:
+        message = Padding(
+            f"[yellow]WARNING some ligands failed to have poses generated see failed_ligands files in [repr.filename]{output_folder}[/repr.filename][/yellow]",
+            (1, 0, 1, 0),
+        )
+        console.print(message)
+    for fail_type, ligands in alchemy_dataset.failed_ligands.items():
```

Comment:
for debugging here, it would be useful to also write out the complex PDB to this same folder. Also useful for checking with the succesfully docked ligands.